### ISSUE TITLE
Flatpak Version - SSH Tunneling through Agent not working

### ISSUE THREAD

**Issue by xaguzman** [dbeaver/dbeaver#9192]
#### System information: 
- Operating system (distribution) and version. Pop_OS 20.04
- DBeaver version   Flatpak ( 7.1.2 )
- Additional extensions 

#### Connection specification:
- Database name and version Posgresql
- Driver name  
- Do you use tunnels or proxies (SSH, SOCKS, etc)?   SSH tunneling

#### Describe the problem you're observing:
While working with the deb package my ssh tunneling works without any problems (I use Agent authentication mode), I can connect fine to my database.
If I switch to the Flatpak version and use the exact same configuration, I keep getting the error:
```
Can't initialize tunnel
  Unable to initialize SSH agent
  Unable to initialize SSH agent
```
Both (deb and flatpak) are version 7.1.2

**Comment by uslss** [comment]
thanks for the bug report

**Comment by xaguzman** [comment]
Thanks for looking into it!
This is the only thing keeping me from going with the flatpak version

**Comment by serge-rider** [comment]
I think it is the same issue as in #8800 . 
The problem is with file access. When a file reside in your user home folder then it is not accessible to dbeaver.

**Comment by riednyko** [comment]
Snap never get access to dotfiles/directories and hidden files for safety.
We sent a request for permission to read such files/directories in the dbeaver snap package. While waiting for a response, to solve this problem, you can move the ssh key out of the /.ssh folder somewhere in homedir, or install the dbeaver-ce snap package using --devmode.

**Comment by riednyko** [comment]
Can`t reproduce this with the flatpak. The same problem exists in the snap version of dbeaver, snap does not give permissions to read the SSH directories. We sent a request for permission to read SSH directories in the dbeaver snap package. While waiting for a response, to solve this problem, the solution for snap may be to install using --devmode.

**Comment by xaguzman** [comment]
are you saying you are able to ssh tunnel using the agent while using the flatpak version? It is still not working for me

**Comment by gryn010** [comment]
I have the same problem on arch linux with dbeaver installed from both the community repo (v7.1.5-1) and aur (v7.2.3-1).
Can't pass the private key directly either since it has no passphrase #10118

**Comment by njenga55** [comment]
os: ubuntu
I had to locate my local installation instance for MySQL, so for my case, I added **/usr/bin** as the home location.

**Comment by uslss** [comment]
is it still actual?

**Comment by uslss** [comment]
there is no update for a long time.
If it is still actual - feel free to reopen the issue.

**Comment by vskult** [comment]
@uslss I am actually facing this issue, but I have a different error message.

I am using the Flatpak version as well and connecting with the agent fails.

With SSHj implementation I get:
```
eclipse.buildId=unknown
java.version=11.0.16.1
java.vendor=Flathub
BootLoader constants: OS=linux, ARCH=x86_64, WS=gtk, NL=en_US
Command-line arguments:  -os linux -ws gtk -arch x86_64

org.jkiss.dbeaver.model
Error
Sat Jan 28 11:32:46 EET 2023
Exhausted available authentication methods

net.schmizz.sshj.userauth.UserAuthException: Exhausted available authentication methods
	at net.schmizz.sshj.SSHClient.auth(SSHClient.java:230)
	at org.jkiss.dbeaver.model.net.ssh.SSHImplementationSshj.setupTunnel(SSHImplementationSshj.java:128)
	at org.jkiss.dbeaver.model.net.ssh.SSHImplementationAbstract.initTunnel(SSHImplementationAbstract.java:134)
	at org.jkiss.dbeaver.model.net.ssh.SSHTunnelImpl.initializeHandler(SSHTunnelImpl.java:77)
	at org.jkiss.dbeaver.ui.net.ssh.SSHTunnelConfiguratorUI$8.execute(SSHTunnelConfiguratorUI.java:307)
	at org.jkiss.dbeaver.ui.net.ssh.SSHTunnelConfiguratorUI$TunnelConnectionTestJob.run(SSHTunnelConfiguratorUI.java:734)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)

```

and with JSch implementation I get the following error:
```
eclipse.buildId=unknown
java.version=11.0.16.1
java.vendor=Flathub
BootLoader constants: OS=linux, ARCH=x86_64, WS=gtk, NL=en_US
Command-line arguments:  -os linux -ws gtk -arch x86_64

org.jkiss.dbeaver.model
Error
Sat Jan 28 11:33:00 EET 2023
Auth fail

com.jcraft.jsch.JSchException: Auth fail
	at com.jcraft.jsch.Session.connect(Session.java:519)
	at com.jcraft.jsch.Session.connect(Session.java:183)
	at org.jkiss.dbeaver.model.net.ssh.SSHImplementationJsch.setupTunnel(SSHImplementationJsch.java:121)
	at org.jkiss.dbeaver.model.net.ssh.SSHImplementationAbstract.initTunnel(SSHImplementationAbstract.java:134)
	at org.jkiss.dbeaver.model.net.ssh.SSHTunnelImpl.initializeHandler(SSHTunnelImpl.java:77)
	at org.jkiss.dbeaver.ui.net.ssh.SSHTunnelConfiguratorUI$8.execute(SSHTunnelConfiguratorUI.java:307)
	at org.jkiss.dbeaver.ui.net.ssh.SSHTunnelConfiguratorUI$TunnelConnectionTestJob.run(SSHTunnelConfiguratorUI.java:734)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)

```

I know my key works as I can use the `Public Key` authentication method and it works, but I'd rather let the keyring save the passphrase for me.  
Also, the key is well loaded in my keyring as I can use `ssh-add -L` to see it, and `ssh`ing into the server works flawlessly.

I also checked the Flatpak permissions using Flatseal and `Secure Shell Agent` is enabled.  
I know a Flatpack should have access to it as KeePassXC can work with it so I figured it is a DBeaver related issue?

**Comment by paulbrzeski** [comment]
Same problem here, updated to latest DBeaver Community 25.1.1 but it's still saying "Error establishing SSHJ tunnel:  Unable to initialize SSH agent"

Running on Linux Mint 22.1 if that helps.

Update: I found a solution! Had to run `dbeaver-ce` from the terminal window where I ran the `ssh-agent`. Looks like it's a scope limitation with how the ssh-agent works. Kudos to https://stackoverflow.com/a/59279911
